automation/package_test/test.py

```diff
@@ -89,7 +97,9 @@ def _run_test(self, test_function, extra, results, test_key):
         try:
             test_function(extra)
         except Exception:
-            results[extra][test_key]["passed"] = False
+            results[extra].setdefault(test_key, {})["passed"] = False
+        else:
+            results[extra].setdefault(test_key, {})["passed"] = True
```

Comment:
I guess you thought about it but why using the else and not just passing the `["passed"] = True` after the `test_function(extra)`

Reply:
I actually didn't, but in general when you wrap with try except you want in the try only the code you expect to fail and want to catch error for, here it's only the `test_function` call, if for example the `extra` key is missing in the `results` dict meaning a `KeyError` will be thrown I don't want to catch that

Reply:
Sababa agreed